{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# reading the list of DEP files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['1.Völgyi_2017/Volgyi_2018_AD_increased_mapped.xlsx\\n', '1.Völgyi_2017/Volgyi_AD_deecreased_mapped.xlsx\\n', '2.Neuner_2017/1-s2.0-S0166432816303564-mmc2_mapped.xlsx\\n', '3.Sweet_2016/diffrential__exp_proteins.xlsx\\n', '5.Chang_2015/CHANG_2015_DES.xlsx\\n', '6.Chang_2013/CHANG_2013_DES_mapped.xlsx\\n', '6.Chang_2013/CHANG_2013_DES_1_mapped.xlsx\\n', '7.Zolochevska_2018/7.Zolochevshka2018_NDANvsAD_DE_mapped_OS.xlsx\\n', '8.Krivinko_2018/Krivinko_2018_PSDEnriched_mice_mapped_OS.xlsx']\n",
      "3.Sweet_2016/diffrential__exp_proteins.xlsx\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Human.Entrez.ID</th>\n",
       "      <th>Human.Gene.Name</th>\n",
       "      <th>Human.Uniprot.ID</th>\n",
       "      <th>Volgyi_2017</th>\n",
       "      <th>Neuner_2017</th>\n",
       "      <th>Sweet_2016</th>\n",
       "      <th>Shi_2014</th>\n",
       "      <th>Chang_2015</th>\n",
       "      <th>Chang_2013</th>\n",
       "      <th>Zolochevska_2018</th>\n",
       "      <th>Krivinko_2018</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>6709</td>\n",
       "      <td>SPTAN1</td>\n",
       "      <td>Q13813</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>6711</td>\n",
       "      <td>SPTBN1</td>\n",
       "      <td>Q01082</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>4741</td>\n",
       "      <td>NEFM</td>\n",
       "      <td>P07197</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>287</td>\n",
       "      <td>ANK2</td>\n",
       "      <td>Q01484</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>1778</td>\n",
       "      <td>DYNC1H1</td>\n",
       "      <td>Q14204</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0  Human.Entrez.ID Human.Gene.Name Human.Uniprot.ID  Volgyi_2017  \\\n",
       "0           0             6709          SPTAN1           Q13813            0   \n",
       "1           1             6711          SPTBN1           Q01082            0   \n",
       "2           2             4741            NEFM           P07197            0   \n",
       "3           3              287            ANK2           Q01484            0   \n",
       "4           4             1778         DYNC1H1           Q14204            0   \n",
       "\n",
       "   Neuner_2017  Sweet_2016  Shi_2014  Chang_2015  Chang_2013  \\\n",
       "0            1           0         0           1           0   \n",
       "1            1           0         0           1           0   \n",
       "2            0           0         0           1           0   \n",
       "3            1           1         0           1           0   \n",
       "4            0           0         0           1           0   \n",
       "\n",
       "   Zolochevska_2018  Krivinko_2018  \n",
       "0                 1              0  \n",
       "1                 0              0  \n",
       "2                 1              0  \n",
       "3                 0              1  \n",
       "4                 0              0  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file = open('DE_protein_filenames.txt','r')\n",
    "# print(file.read())\n",
    "filelist = file.readlines()\n",
    "file.close()\n",
    "\n",
    "print(filelist)\n",
    "print(filelist[3])\n",
    "\n",
    "files_to_map = [3, 4]\n",
    "\n",
    "list_of_all = pd.read_excel('try2.xlsx')\n",
    "list_of_all.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CALB2 : 0\n",
      "SEPT1 : 0\n",
      "DCTN : 0\n",
      "HSP90A : 0\n",
      "CALB2 : 0\n",
      "NELE : 1\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "DE_proteins = [[],[], [], [], [], [], [], [], [] ]\n",
    "\n",
    "protein_col_names = ['Protein', 'Gene name']\n",
    "skip_first_line =[ [], [1]]\n",
    "skip_first_line = [ [0,2],[] ,[],[],[1],[],[],[], []]\n",
    "\n",
    "for i in range(len(files_to_map)):\n",
    "    n = files_to_map[i]\n",
    "    file_name = filelist[n]\n",
    "    pro_col = protein_col_names[i]\n",
    "    file_name = file_name.split('\\n')[0]\n",
    "    dir_name = file_name.split('/')[0]\n",
    "    e_list = []\n",
    "#     print(dir_name)\n",
    "#     print(file_name)\n",
    "#     print(os.getcwd())\n",
    "#     print(os.listdir(dir_name))\n",
    "#     file_name = '3.Sweet_2016\\diffrential_exp_proteins.xlsx'\n",
    "#     diffrential_exp_proteins.xlsx\n",
    "# if(skip)\n",
    "    thefile = pd.read_excel(file_name, skiprows=skip_first_line[n])\n",
    "    for j in range(len(thefile)):\n",
    "#         print(thefile[pro_col][i])\n",
    "        Entrez_id = list_of_all.loc[list_of_all['Human.Gene.Name'] == thefile[pro_col][j] ]\n",
    "#         print(len(Entrez_id))\n",
    "        if(len(Entrez_id)):\n",
    "            ind = Entrez_id.index\n",
    "            Entrez_id = Entrez_id['Human.Entrez.ID'][ind[0]]\n",
    "            e_list.append(Entrez_id)\n",
    "        else:\n",
    "            print(thefile[pro_col][j] + ' : ' + str(i))\n",
    "#     print(thefile[pro_col])\n",
    "    DE_proteins[n].append(e_list)\n",
    "    \n",
    "#     filename = file_name.split('\\n')\n",
    "#     print(filename[0])\n",
    "#     print(thefile.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "a = list_of_all.loc[list_of_all['Human.Gene.Name'] == 'SEPT1']\n",
    "b = a.index\n",
    "# b[0]\n",
    "# a['Human.Entrez.ID'][b[0]]\n",
    "# a\n",
    "\n",
    "# DE_proteins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# the other papers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----\n",
      " 0\n",
      "1.Völgyi_2017/Volgyi_2018_AD_increased_mapped.xlsx\n",
      "['proteins(geneids).xlsx', 'Volgyi_2018_AD_increased.xlsx', 'Volgyi_2018_AD_increased_mapped.xlsx', 'Volgyi_AD_deecreased.xlsx', 'Volgyi_AD_deecreased_mapped.xlsx', 'Völgyi2018_Article_ChronicCerebralHypoperfusionIn.pdf', 'with_proteinNames_and_fucntion.xlsx']\n",
      "[0, 2]\n",
      "----\n",
      " 1\n",
      "1.Völgyi_2017/Volgyi_AD_deecreased_mapped.xlsx\n",
      "['proteins(geneids).xlsx', 'Volgyi_2018_AD_increased.xlsx', 'Volgyi_2018_AD_increased_mapped.xlsx', 'Volgyi_AD_deecreased.xlsx', 'Volgyi_AD_deecreased_mapped.xlsx', 'Völgyi2018_Article_ChronicCerebralHypoperfusionIn.pdf', 'with_proteinNames_and_fucntion.xlsx']\n",
      "[]\n",
      "----\n",
      " 2\n",
      "2.Neuner_2017/1-s2.0-S0166432816303564-mmc2_mapped.xlsx\n",
      "['1-s2.0-S0166432816303564-main.pdf', '1-s2.0-S0166432816303564-mmc2.xlsx', '1-s2.0-S0166432816303564-mmc2_mapped - Copy.xlsx', '1-s2.0-S0166432816303564-mmc2_mapped.xlsb', '1-s2.0-S0166432816303564-mmc2_mapped.xlsx', 'proteins.xlsx']\n",
      "[]\n",
      "----\n",
      " 5\n",
      "6.Chang_2013/CHANG_2013_DES_mapped.xlsx\n",
      "['1-s2.0-S1552526012023862-main.pdf', 'CHANG_2013_DES.xlsx', 'Chang_2013_DES_1.xlsx', 'Chang_2013_DES_1_mapped.xlsx', 'CHANG_2013_DES_mapped.xlsx', 'proteins.xlsx']\n",
      "[]\n",
      "----\n",
      " 6\n",
      "6.Chang_2013/CHANG_2013_DES_1_mapped.xlsx\n",
      "['1-s2.0-S1552526012023862-main.pdf', 'CHANG_2013_DES.xlsx', 'Chang_2013_DES_1.xlsx', 'Chang_2013_DES_1_mapped.xlsx', 'CHANG_2013_DES_mapped.xlsx', 'proteins.xlsx']\n",
      "[]\n",
      "----\n",
      " 7\n",
      "7.Zolochevska_2018/7.Zolochevshka2018_NDANvsAD_DE_mapped_OS.xlsx\n",
      "['7.Zolochevshka2018_NDANvsAD_DE.xlsx', '7.Zolochevshka2018_NDANvsAD_DE_mapped.xlsx', '7.Zolochevshka2018_NDANvsAD_DE_mapped_OS.xlsx', 'proteins(geneid_assensionnos).xlsx', 'Zolochevska et al. - 2018 - Postsynaptic Proteome of Non-Demented Individuals .pdf']\n",
      "[]\n",
      "----\n",
      " 8\n",
      "8.Krivinko_2018/Krivinko_2018_PSDEnriched_mice_mapped_OS.xlsx\n",
      "['appi.ajp.2018.17080858.ds004.Table_S6_mapped.xlsx', 'Krivinko et al. - 2018 - Synaptic Proteome Compensation and Resilience to P.pdf', 'Krivinko_2018_ADwithvswithoutpsy.xlsx', 'Krivinko_2018_ADwithvswithoutpsy_mapped.xlsx', 'Krivinko_2018_ADwithvswithoutpsy_mapped_OS.xlsx', 'Krivinko_2018_PSDEnriched_mice.xlsx', 'Krivinko_2018_PSDEnriched_mice_mapped.xlsx', 'Krivinko_2018_PSDEnriched_mice_mapped_OS.xlsx', 'proteins.xlsx']\n",
      "[]\n",
      "   Protein  PSD level : Homogenate level  Entrez_id Gene name\n",
      "0   SRCIN1                      7.176861    80725.0     80725\n",
      "1  SYNGAP1                      6.570652     8831.0      8831\n",
      "2    GRIA3                      6.371791     2892.0      2892\n",
      "3    GRIA4                      6.329123     2893.0      2893\n",
      "4   HOMER1                      6.318000     9456.0      9456\n"
     ]
    }
   ],
   "source": [
    "\n",
    "Entrez_col_names = [ 'Entrez ID Human', 'Entrez ID Human', 'Entrez ID Human', 'Entrez ID Human', 'Entrez ID Human', 'Entrez ID Human','Entrez ID Human', 'Entrez_id', 'Entrez_id' ] \n",
    "\n",
    "done = [ 0, 1, 2, 5, 6,7]\n",
    "\n",
    "dts = {\n",
    "    'Entrez ID Human': int,\n",
    "    'Entrez_id': int\n",
    "}\n",
    "\n",
    "for i in range( len(DE_proteins) ):\n",
    "    if( (i not in files_to_map) ):\n",
    "        print('----\\n',i)\n",
    "        file_name = filelist[i]\n",
    "        file_name = file_name.split('\\n')[0]\n",
    "        print(file_name)\n",
    "        dir_name = file_name.split('/')[0]\n",
    "        print(os.listdir(dir_name))\n",
    "#         file_name = '1.Völgyi_2017/Volgyi_2018_AD_increased_mapped.xlsx'\n",
    "#                                    Volgyi_2018_AD_increased_mapped.xlsx\n",
    "        print(skip_first_line[i])\n",
    "        thefile = pd.read_excel(file_name, skiprows=skip_first_line[i])#, dtype=dts)\n",
    "#         print(thefile['Entrez ID Human'])\n",
    "        DE_proteins[i].append(thefile[Entrez_col_names[i]])\n",
    "        if( i not in done):\n",
    "            print(thefile.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      2617.0\n",
       "1      3190.0\n",
       "2      4817.0\n",
       "3      3094.0\n",
       "4      1653.0\n",
       "5     10130.0\n",
       "6         NaN\n",
       "7      9318.0\n",
       "8      5682.0\n",
       "9      5702.0\n",
       "10     5704.0\n",
       "11        NaN\n",
       "12     8574.0\n",
       "13      586.0\n",
       "14    55163.0\n",
       "15        NaN\n",
       "16     5049.0\n",
       "17      348.0\n",
       "18        NaN\n",
       "19     2819.0\n",
       "20        NaN\n",
       "21     8682.0\n",
       "22     4199.0\n",
       "23        NaN\n",
       "24     3251.0\n",
       "25        NaN\n",
       "26     5604.0\n",
       "27        NaN\n",
       "28     7018.0\n",
       "29      794.0\n",
       "30        NaN\n",
       "31     1428.0\n",
       "Name: Entrez ID Human, dtype: float64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p = '2617'\n",
    "if(p in DE_proteins[6][0]):\n",
    "    print('adfdaf')\n",
    "DE_proteins[0][0]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# bringing them together into a df showing which is in which"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "411\n",
      "9\n",
      "9\n",
      "False\n",
      "True\n"
     ]
    }
   ],
   "source": [
    "# creating a single list with all of them\n",
    "\n",
    "DE_p_2 = [[],[], [], [], [], [], [], [], [] ]\n",
    "import math\n",
    "DE_proteins_unique = []\n",
    "for i in range(len(DE_proteins)):\n",
    "    for j in range(len(DE_proteins[i][0])):\n",
    "        if(not (math.isnan(DE_proteins[i][0][j]))):\n",
    "            DE_p_2[i].append(str(int(DE_proteins[i][0][j])))\n",
    "            if( DE_proteins[i][0][j] not in DE_proteins_unique):\n",
    "                DE_proteins_unique.append(str(int(DE_proteins[i][0][j])))\n",
    "#                 DE_proteins_unique.append(DE_proteins[i][0][j])\n",
    "                \n",
    "#             DE_proteins[i][0][j] = str(int(DE_proteins[i][0][j]))\n",
    "\n",
    "                #           DE_  print(str(int(DE_proteins[i][0][j])))\n",
    "\n",
    "print(len(DE_proteins_unique))\n",
    "# print(DE_proteins_unique)\n",
    "\n",
    "\n",
    "# checking in each making df\n",
    "\n",
    "\n",
    "colnames = ['Entrez ID']\n",
    "\n",
    "for i in range(len(filelist)):\n",
    "    name = filelist[i].split('\\n')[0]\n",
    "    colname = name.split('/')[0]\n",
    "    if(colname not in colnames):\n",
    "        colnames.append(colname)\n",
    "\n",
    "# print(colnames)\n",
    "print(len(filelist))\n",
    "print(len(DE_proteins))\n",
    "\n",
    "DE_df = pd.DataFrame(columns=colnames)\n",
    "\n",
    "for i in range(len(DE_proteins_unique)):\n",
    "    pro = DE_proteins_unique[i]\n",
    "    newrow = {\n",
    "        'Entrez ID': pro,\n",
    "        '1.Völgyi_2017': 0,\n",
    "        '2.Neuner_2017': 0,\n",
    "        '3.Sweet_2016': 0,\n",
    "        '5.Chang_2015': 0,\n",
    "        '6.Chang_2013': 0,\n",
    "        '7.Zolochevska_2018': 0,\n",
    "        '8.Krivinko_2018': 0\n",
    "    }\n",
    "    DE_df = DE_df.append(newrow,ignore_index=True)\n",
    "    for j in range(len(filelist)):\n",
    "        name = filelist[j].split('\\n')[0]\n",
    "        colname = name.split('/')[0]\n",
    "#         print(colname)\n",
    "#         print(DE_proteins[j][0])\n",
    "#         print(pro, int(pro) in DE_proteins[j][0], j, type(pro))\n",
    "        if(pro in DE_p_2[j]):\n",
    "#             if(j == 0):\n",
    "#                 print('in')\n",
    "#             print( pro, ' ', colname)\n",
    "            DE_df.loc[DE_df['Entrez ID'] == pro, colname] = 1\n",
    "\n",
    "\n",
    "DE_df['count'] = DE_df['1.Völgyi_2017'] + DE_df['2.Neuner_2017'] + DE_df['3.Sweet_2016'] + DE_df['5.Chang_2015'] + DE_df['6.Chang_2013'] + DE_df['7.Zolochevska_2018'] + DE_df['8.Krivinko_2018']\n",
    "\n",
    "\n",
    "print( '2167' in DE_proteins_unique)\n",
    "print( '2617' in DE_proteins_unique)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 326,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2617.0\n",
      "False\n",
      "True\n"
     ]
    }
   ],
   "source": [
    "p = float(2617)\n",
    "print(p)\n",
    "print( p in DE_proteins[0][0])\n",
    "print( 2617.0 == DE_proteins[0][0][0])\n",
    "# DE_df.loc[DE_df['Entrez ID'] == '2617','2.Neuner_2017' ] = 1\n",
    "# DE_proteins_unique\n",
    "DE_df.head()\n",
    "# DE_proteins[0][0]\n",
    "# DE_df.sort_values('count', ascending=False)\n",
    "DE_df.to_csv('Diffrentially_Expressed.csv',index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "done\n"
     ]
    }
   ],
   "source": [
    "DEproteinslist = []\n",
    "DEproteinslist = pd.DataFrame()\n",
    "DEproteinslist['Entrez id'] = DE_df['Entrez ID']\n",
    "DEproteinslist['A'] = 'Diff_ex'\n",
    "DEproteinslist['B'] = 'Differentiallly expressed proteins'\n",
    "# print(len(DEproteinslist))\n",
    "# print(len(DE_df['Entrez ID']))\n",
    "# print(len(DE_df))\n",
    "# DEproteinslist\n",
    "# DE_df.head(450)\n",
    "# print(DEproteinslist)\n",
    "# DEproteinslist.to_csv('flatfile.DE.csv',sep='\\t',header=False, columns=['A','B','Entrez id'],index=False)\n",
    "print('done')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Entrez ID' '1.Völgyi_2017' '2.Neuner_2017' '3.Sweet_2016' '5.Chang_2015'\n",
      " '6.Chang_2013' '7.Zolochevska_2018' '8.Krivinko_2018' 'count']\n"
     ]
    }
   ],
   "source": [
    "DEproteins3list = []\n",
    "DEproteins3 = pd.DataFrame()\n",
    "DEproteins_3 = pd.DataFrame()\n",
    "\n",
    "DEproteins2list = []\n",
    "DEproteins2 = pd.DataFrame()\n",
    "DEproteins_2 = pd.DataFrame()\n",
    "\n",
    "DE_df.sort_values(by='count')\n",
    "print(DE_df.columns.values)\n",
    "DEproteins  = DE_df[['Entrez ID','count']]\n",
    "DEproteins3 = DEproteins[DEproteins['count'] == 3]\n",
    "DEproteins_3['Entrez id'] = DEproteins3['Entrez ID']\n",
    "DEproteins_3['A'] = 'Diff_ex_3'\n",
    "DEproteins_3['B'] = 'Differentially expressed proteins in 3'\n",
    "# DEproteins3['Entrez id'] == \n",
    "\n",
    "DEproteins2 = DEproteins[DEproteins['count'] == 2]\n",
    "DEproteins_2['Entrez id'] = DEproteins2['Entrez ID']\n",
    "DEproteins_2['A'] = 'Diff_ex_2'\n",
    "DEproteins_2['B'] = 'Differentially expressed proteins in 2'\n",
    "\n",
    "DEproteins_2.head()\n",
    "\n",
    "DEproteinslist = DEproteinslist.append(DEproteins_3,ignore_index=True )\n",
    "DEproteinslist = DEproteinslist.append(DEproteins_2,ignore_index=True )\n",
    "DEproteinslist['A'].unique()\n",
    "# DEproteinslist\n",
    "\n",
    "DEproteinslist.to_csv('flatfile.DE.sep.csv',sep='\\t',header=False, columns=['A','B','Entrez id'],index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "finalproject",
   "language": "python",
   "name": "finalproject"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
