{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# reading the list of DEP files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['1.Völgyi_2017/Volgyi_2018_AD_increased_mapped.xlsx\\n', '1.Völgyi_2017/Volgyi_AD_deecreased_mapped.xlsx\\n', '2.Neuner_2017/1-s2.0-S0166432816303564-mmc2_mapped.xlsx\\n', '3.Sweet_2016/diffrential__exp_proteins.xlsx\\n', '5.Chang_2015/CHANG_2015_DES.xlsx\\n', '6.Chang_2013/CHANG_2013_DES_mapped.xlsx\\n', '6.Chang_2013/CHANG_2013_DES_1_mapped.xlsx\\n', '7.Zolochevska_2018/7.Zolochevshka2018_NDANvsAD_DE_mapped_OS.xlsx\\n', '8.Krivinko_2018/Krivinko_2018_PSDEnriched_mice_mapped_OS.xlsx']\n",
      "3.Sweet_2016/diffrential__exp_proteins.xlsx\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Human.Entrez.ID</th>\n",
       "      <th>Human.Gene.Name</th>\n",
       "      <th>Human.Uniprot.ID</th>\n",
       "      <th>Volgyi_2017</th>\n",
       "      <th>Neuner_2017</th>\n",
       "      <th>Sweet_2016</th>\n",
       "      <th>Shi_2014</th>\n",
       "      <th>Chang_2015</th>\n",
       "      <th>Chang_2013</th>\n",
       "      <th>Zolochevska_2018</th>\n",
       "      <th>Krivinko_2018</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>6709</td>\n",
       "      <td>SPTAN1</td>\n",
       "      <td>Q13813</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>6711</td>\n",
       "      <td>SPTBN1</td>\n",
       "      <td>Q01082</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>4741</td>\n",
       "      <td>NEFM</td>\n",
       "      <td>P07197</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>287</td>\n",
       "      <td>ANK2</td>\n",
       "      <td>Q01484</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>1778</td>\n",
       "      <td>DYNC1H1</td>\n",
       "      <td>Q14204</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 0  Human.Entrez.ID Human.Gene.Name Human.Uniprot.ID  Volgyi_2017  \\\n",
       "0           0             6709          SPTAN1           Q13813            0   \n",
       "1           1             6711          SPTBN1           Q01082            0   \n",
       "2           2             4741            NEFM           P07197            0   \n",
       "3           3              287            ANK2           Q01484            0   \n",
       "4           4             1778         DYNC1H1           Q14204            0   \n",
       "\n",
       "   Neuner_2017  Sweet_2016  Shi_2014  Chang_2015  Chang_2013  \\\n",
       "0            1           0         0           1           0   \n",
       "1            1           0         0           1           0   \n",
       "2            0           0         0           1           0   \n",
       "3            1           1         0           1           0   \n",
       "4            0           0         0           1           0   \n",
       "\n",
       "   Zolochevska_2018  Krivinko_2018  \n",
       "0                 1              0  \n",
       "1                 0              0  \n",
       "2                 1              0  \n",
       "3                 0              1  \n",
       "4                 0              0  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "file = open('DE_protein_filenames.txt','r')\n",
    "# print(file.read())\n",
    "filelist = file.readlines()\n",
    "file.close()\n",
    "\n",
    "print(filelist)\n",
    "print(filelist[3])\n",
    "\n",
    "files_to_map = [3, 4]\n",
    "\n",
    "list_of_all = pd.read_excel('try2.xlsx')\n",
    "list_of_all.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CALB2 : 0\n",
      "SEPT1 : 0\n",
      "DCTN : 0\n",
      "HSP90A : 0\n",
      "CALB2 : 0\n",
      "[10801, 119, 4133, 2319, 112755, 11069, 5052, 10211, 9554, 2776, 3306, 5962, 9228, 8803, 11069, 2912, 4133, 85358, 9228, 2913, 2904, 2892, 9456, 4724, 2891, 2893, 5052, 2776, 7431, 10801, 5962, 387, 5478, 3306, 2597, 8803, 388, 1267, 119, 1808, 203, 1173, 112755, 10971, 10728, 998, 5898, 1267, 4130, 4133, 7431, 478, 4724, 119, 2319, 4735, 10211, 112755, 6844, 3208, 23154]\n",
      "NELE : 1\n",
      "[9118, 6900, 1462, 30845, 4744, 4741, 5093, 51645, 7001, 5694, 391, 4735, 5413, 989, 23176, 10801, 55752, 80725, 6854, 7277, 10382, 7781, 11345, 23530, 10493, 22884, 11076, 22895, 1743]\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "DE_proteins = [[],[], [], [], [], [], [], [], [] ]\n",
    "\n",
    "protein_col_names = ['Protein', 'Gene name']\n",
    "skip_first_line =[ [], [1]]\n",
    "skip_first_line = [ [0,2],[] ,[],[],[1],[],[],[], []]\n",
    "\n",
    "for i in range(len(files_to_map)):\n",
    "    n = files_to_map[i]\n",
    "    file_name = filelist[n]\n",
    "    pro_col = protein_col_names[i]\n",
    "    file_name = file_name.split('\\n')[0]\n",
    "    dir_name = file_name.split('/')[0]\n",
    "    e_list = []\n",
    "#     print(dir_name)\n",
    "#     print(file_name)\n",
    "#     print(os.getcwd())\n",
    "#     print(os.listdir(dir_name))\n",
    "#     file_name = '3.Sweet_2016\\diffrential_exp_proteins.xlsx'\n",
    "#     diffrential_exp_proteins.xlsx\n",
    "# if(skip)\n",
    "    thefile = pd.read_excel(file_name, skiprows=skip_first_line[n])\n",
    "    for j in range(len(thefile)):\n",
    "#         print(thefile[pro_col][i])\n",
    "        Entrez_id = list_of_all.loc[list_of_all['Human.Gene.Name'] == thefile[pro_col][j] ]\n",
    "#         print(len(Entrez_id))\n",
    "        if(len(Entrez_id)):\n",
    "            ind = Entrez_id.index\n",
    "            Entrez_id = Entrez_id['Human.Entrez.ID'][ind[0]]\n",
    "            e_list.append(Entrez_id)\n",
    "        else:\n",
    "            print(thefile[pro_col][j] + ' : ' + str(i))\n",
    "#     print(thefile[pro_col])\n",
    "    print(e_list)\n",
    "    DE_proteins[n].append(np.array(e_list, dtype=int))\n",
    "    \n",
    "#     filename = file_name.split('\\n')\n",
    "#     print(filename[0])\n",
    "#     print(thefile.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ 10801,    119,   4133,   2319, 112755,  11069,   5052,  10211,\n",
       "         9554,   2776,   3306,   5962,   9228,   8803,  11069,   2912,\n",
       "         4133,  85358,   9228,   2913,   2904,   2892,   9456,   4724,\n",
       "         2891,   2893,   5052,   2776,   7431,  10801,   5962,    387,\n",
       "         5478,   3306,   2597,   8803,    388,   1267,    119,   1808,\n",
       "          203,   1173, 112755,  10971,  10728,    998,   5898,   1267,\n",
       "         4130,   4133,   7431,    478,   4724,    119,   2319,   4735,\n",
       "        10211, 112755,   6844,   3208,  23154])"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = list_of_all.loc[list_of_all['Human.Gene.Name'] == 'SEPT1']\n",
    "b = a.index\n",
    "# b[0]\n",
    "# a['Human.Entrez.ID'][b[0]]\n",
    "# a\n",
    "DE_proteins[3][0]\n",
    "# DE_proteins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# the other papers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----\n",
      " 0\n",
      "1.Völgyi_2017/Volgyi_2018_AD_increased_mapped.xlsx\n",
      "['proteins(geneids).xlsx', 'Volgyi_2018_AD_increased.xlsx', 'Volgyi_2018_AD_increased_mapped.xlsx', 'Volgyi_AD_deecreased.xlsx', 'Volgyi_AD_deecreased_mapped.xlsx', 'Völgyi2018_Article_ChronicCerebralHypoperfusionIn.pdf', 'with_proteinNames_and_fucntion.xlsx']\n",
      "[0, 2]\n",
      "32\n",
      "23\n",
      "[ 2617  3190  4817  3094  1653 10130  9318  5682  5702  5704  8574   586\n",
      " 55163  5049   348  2819  8682  4199  3251  5604  7018   794  1428]\n",
      "----\n",
      " 1\n",
      "1.Völgyi_2017/Volgyi_AD_deecreased_mapped.xlsx\n",
      "['proteins(geneids).xlsx', 'Volgyi_2018_AD_increased.xlsx', 'Volgyi_2018_AD_increased_mapped.xlsx', 'Volgyi_AD_deecreased.xlsx', 'Volgyi_AD_deecreased_mapped.xlsx', 'Völgyi2018_Article_ChronicCerebralHypoperfusionIn.pdf', 'with_proteinNames_and_fucntion.xlsx']\n",
      "[]\n",
      "32\n",
      "23\n",
      "[  6843   6616 112755  22848   9522  11346  22941  50944   9568   6529\n",
      "   6506    490    477    476    478   5906   2697  10675   4340   4978\n",
      "    961    966  10452]\n",
      "----\n",
      " 2\n",
      "2.Neuner_2017/1-s2.0-S0166432816303564-mmc2_mapped.xlsx\n",
      "['1-s2.0-S0166432816303564-main.pdf', '1-s2.0-S0166432816303564-mmc2.xlsx', '1-s2.0-S0166432816303564-mmc2_mapped - Copy.xlsx', '1-s2.0-S0166432816303564-mmc2_mapped.xlsb', '1-s2.0-S0166432816303564-mmc2_mapped.xlsx', 'proteins.xlsx']\n",
      "[]\n",
      "103\n",
      "103\n",
      "[    50    134    287    161    163    348    351  55102    479    481\n",
      "    493    527    526    529   9212    622  10487  84303  54918  28958\n",
      "   1329   1345 140690   1737   1743  10059   1938   2554   2596   2670\n",
      "   2746   2773  54331   2806   2823   2824   2861   2890   2913   2944\n",
      "   5095 348980   7923   3312  93185  10989   3615   9922   3708   3785\n",
      "   3848   3858   3849   4054   4099   5594  54996   4172   4191  25902\n",
      "   4540   2475  91663  23154 126328   4700   4701   4884   4897   9369\n",
      "  55753  51090   5621   5802   9230  22931 115827   5898   6184   6252\n",
      "  57142   9522   9554   6563  25769  79751   6512 219537   6616   6709\n",
      "   6711  80725   8803   6853   6854   8831   9145   6950 254863   9804\n",
      " 286262   7284  55737]\n",
      "----\n",
      " 5\n",
      "6.Chang_2013/CHANG_2013_DES_mapped.xlsx\n",
      "['1-s2.0-S1552526012023862-main.pdf', 'CHANG_2013_DES.xlsx', 'Chang_2013_DES_1.xlsx', 'Chang_2013_DES_1_mapped.xlsx', 'CHANG_2013_DES_mapped.xlsx', 'proteins.xlsx']\n",
      "[]\n",
      "59\n",
      "53\n",
      "[ 5223  2664  1152 30845  1808 23176  2746  2023  2752    60  4190  7846\n",
      " 92399  7345  5052   396  9615 23176  2026  2752  5223  1152  2805  5315\n",
      "    60  2026  4190  7846  5052   396  3312  3312 10015  1400   230  2664\n",
      " 26227   396 80700  5019 23176 55752  2775  2775  5315 26227  2775   396\n",
      "  3312  3312  9615 80700 11076]\n",
      "----\n",
      " 6\n",
      "6.Chang_2013/CHANG_2013_DES_1_mapped.xlsx\n",
      "['1-s2.0-S1552526012023862-main.pdf', 'CHANG_2013_DES.xlsx', 'Chang_2013_DES_1.xlsx', 'Chang_2013_DES_1_mapped.xlsx', 'CHANG_2013_DES_mapped.xlsx', 'proteins.xlsx']\n",
      "[]\n",
      "24\n",
      "18\n",
      "[ 2665    60  6857   308    60 22933  2665    60  2023   308  3315    60\n",
      "  2946 10935  2023   523   523  1400]\n",
      "----\n",
      " 7\n",
      "7.Zolochevska_2018/7.Zolochevshka2018_NDANvsAD_DE_mapped_OS.xlsx\n",
      "['7.Zolochevshka2018_NDANvsAD_DE.xlsx', '7.Zolochevshka2018_NDANvsAD_DE_mapped.xlsx', '7.Zolochevshka2018_NDANvsAD_DE_mapped_OS.xlsx', 'proteins(geneid_assensionnos).xlsx', 'Zolochevska et al. - 2018 - Postsynaptic Proteome of Non-Demented Individuals .pdf']\n",
      "[]\n",
      "63\n",
      "63\n",
      "[   71   302   815   811  1152  1152  2670  2670  2670  2670  2670  2670\n",
      "  2670  2670  2670  2670  2597  3043  3043  3043  7414  2670  2670  2670\n",
      "  2670  2670  2670  2670  2670  1759  1759 25824  2011  1267 25824  6853\n",
      "  6853  1211  3858  3857  3857  3848  3848  3848  4191  4698  4723  4741\n",
      "  5217  5217  5091   396   989  6709  6812  7846 10376 10376 10376  7280\n",
      " 10382 84617  7345]\n",
      "----\n",
      " 8\n",
      "8.Krivinko_2018/Krivinko_2018_PSDEnriched_mice_mapped_OS.xlsx\n",
      "['appi.ajp.2018.17080858.ds004.Table_S6_mapped.xlsx', 'Krivinko et al. - 2018 - Synaptic Proteome Compensation and Resilience to P.pdf', 'Krivinko_2018_ADwithvswithoutpsy.xlsx', 'Krivinko_2018_ADwithvswithoutpsy_mapped.xlsx', 'Krivinko_2018_ADwithvswithoutpsy_mapped_OS.xlsx', 'Krivinko_2018_PSDEnriched_mice.xlsx', 'Krivinko_2018_PSDEnriched_mice_mapped.xlsx', 'Krivinko_2018_PSDEnriched_mice_mapped_OS.xlsx', 'proteins.xlsx']\n",
      "[]\n",
      "39\n",
      "38\n",
      "[    80725      8831      2892      2893      9456      2890      1742\n",
      "      5682      2891       815       287       816      8997       286\n",
      "      1499      6262     11331      8402      5245      6708      6853\n",
      "      6710      8573     10801      5582       535 100526833     55964\n",
      "     79751      3708      6543       292      4130      4735     57468\n",
      "      2915      7417      2770]\n",
      "     Protein  PSD level : Homogenate level  Entrez_id Gene name\n",
      "0     SRCIN1                      7.176861    80725.0     80725\n",
      "1    SYNGAP1                      6.570652     8831.0      8831\n",
      "2      GRIA3                      6.371791     2892.0      2892\n",
      "3      GRIA4                      6.329123     2893.0      2893\n",
      "4     HOMER1                      6.318000     9456.0      9456\n",
      "5      GRIA1                      6.224372     2890.0      2890\n",
      "6       DLG4                      5.622708     1742.0      1742\n",
      "7      PSMA1                      5.399001     5682.0      5682\n",
      "8      GRIA2                      5.235197     2891.0      2891\n",
      "9    CAMKIIA                      5.233618      815.0    CAMK2A\n",
      "10      ANK2                      5.163812      287.0      ANK2\n",
      "11   CAMKIIB                      4.944273      816.0    CAMK2B\n",
      "12     KALRN                      4.642313     8997.0     KALRN\n",
      "13      ANK1                      3.799119      286.0      ANK1\n",
      "14    CTNNB1                      3.725078     1499.0    CTNNB1\n",
      "15      RYR2                      3.408028     6262.0      RYR2\n",
      "16      PHB2                      3.245276    11331.0      PHB2\n",
      "18  SLC25A11                      3.186274     8402.0  SLC25A11\n",
      "19       PHB                      3.180728     5245.0       PHB\n",
      "20     SPTA1                      3.097256     6708.0     SPTA1\n"
     ]
    }
   ],
   "source": [
    "\n",
    "Entrez_col_names = [ 'Entrez ID Human', 'Entrez ID Human', 'Entrez ID Human', 'Entrez ID Human', 'Entrez ID Human', 'Entrez ID Human','Entrez ID Human', 'Entrez_id', 'Entrez_id' ] \n",
    "\n",
    "done = [ 0, 1, 2, 5, 6,7]\n",
    "\n",
    "dts = {\n",
    "    'Entrez ID Human': int,\n",
    "    'Entrez_id': int\n",
    "}\n",
    "\n",
    "for i in range( len(DE_proteins) ):\n",
    "    if( (i not in files_to_map) ):\n",
    "        print('----\\n',i)\n",
    "        file_name = filelist[i]\n",
    "        file_name = file_name.split('\\n')[0]\n",
    "        print(file_name)\n",
    "        dir_name = file_name.split('/')[0]\n",
    "        print(os.listdir(dir_name))\n",
    "#         file_name = '1.Völgyi_2017/Volgyi_2018_AD_increased_mapped.xlsx'\n",
    "#                                    Volgyi_2018_AD_increased_mapped.xlsx\n",
    "        print(skip_first_line[i])\n",
    "        thefile = pd.read_excel(file_name, skiprows=skip_first_line[i])#, dtype=dts)\n",
    "        print(len(thefile))\n",
    "        thefile = thefile.dropna(subset=[Entrez_col_names[i]])\n",
    "        print(len(thefile))\n",
    "        lisst = np.array(list(thefile[Entrez_col_names[i]]), dtype=int)\n",
    "        \n",
    "#         print(thefile[Entrez_col_names[i]])\n",
    "        print(lisst)\n",
    "        DE_proteins[i].append(np.array(list(thefile[Entrez_col_names[i]]), dtype=int))\n",
    "        if( i not in done):\n",
    "            print(thefile.head(20))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ 2617,  3190,  4817,  3094,  1653, 10130,  9318,  5682,  5702,\n",
       "        5704,  8574,   586, 55163,  5049,   348,  2819,  8682,  4199,\n",
       "        3251,  5604,  7018,   794,  1428])"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# p = '2617'\n",
    "# if(p in DE_proteins[6][0]):\n",
    "#     print('adfdaf')\n",
    "DE_proteins[0][0]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# bringing them together into a df showing which is in which"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "276\n",
      "9\n",
      "9\n",
      "93185   2 2.Neuner_2017\n",
      "4099   2 2.Neuner_2017\n",
      "523   6 6.Chang_2013\n",
      "9228   3 3.Sweet_2016\n",
      "526   2 2.Neuner_2017\n",
      "527   2 2.Neuner_2017\n",
      "91663   2 2.Neuner_2017\n",
      "529   2 2.Neuner_2017\n",
      "9230   2 2.Neuner_2017\n",
      "5091   7 7.Zolochevska_2018\n",
      "3094   0 1.Völgyi_2017\n",
      "535   8 8.Krivinko_2018\n",
      "3615   2 2.Neuner_2017\n",
      "4130   3 3.Sweet_2016\n",
      "4130   8 8.Krivinko_2018\n",
      "2596   2 2.Neuner_2017\n",
      "4133   3 3.Sweet_2016\n",
      "2597   3 3.Sweet_2016\n",
      "2597   7 7.Zolochevska_2018\n",
      "6184   2 2.Neuner_2017\n",
      "10801   3 3.Sweet_2016\n",
      "10801   4 5.Chang_2015\n",
      "10801   8 8.Krivinko_2018\n",
      "5682   0 1.Völgyi_2017\n",
      "5682   8 8.Krivinko_2018\n",
      "50   2 2.Neuner_2017\n",
      "6708   8 8.Krivinko_2018\n",
      "6709   2 2.Neuner_2017\n",
      "6709   7 7.Zolochevska_2018\n",
      "286262   2 2.Neuner_2017\n",
      "6711   2 2.Neuner_2017\n",
      "6710   8 8.Krivinko_2018\n",
      "2617   0 1.Völgyi_2017\n",
      "54331   2 2.Neuner_2017\n",
      "60   5 6.Chang_2013\n",
      "60   6 6.Chang_2013\n",
      "5694   4 5.Chang_2015\n",
      "11331   8 8.Krivinko_2018\n",
      "5702   0 1.Völgyi_2017\n",
      "71   7 7.Zolochevska_2018\n",
      "5704   0 1.Völgyi_2017\n",
      "586   0 1.Völgyi_2017\n",
      "4172   2 2.Neuner_2017\n",
      "9804   2 2.Neuner_2017\n",
      "11345   4 5.Chang_2015\n",
      "11346   1 1.Völgyi_2017\n",
      "4698   7 7.Zolochevska_2018\n",
      "4700   2 2.Neuner_2017\n",
      "4701   2 2.Neuner_2017\n",
      "4190   5 6.Chang_2013\n",
      "4191   2 2.Neuner_2017\n",
      "4191   7 7.Zolochevska_2018\n",
      "5217   7 7.Zolochevska_2018\n",
      "8803   2 2.Neuner_2017\n",
      "8803   3 3.Sweet_2016\n",
      "7781   4 5.Chang_2015\n",
      "9318   0 1.Völgyi_2017\n",
      "4199   0 1.Völgyi_2017\n",
      "5223   5 6.Chang_2013\n",
      "2664   5 6.Chang_2013\n",
      "2665   6 6.Chang_2013\n",
      "6252   2 2.Neuner_2017\n",
      "7277   4 5.Chang_2015\n",
      "622   2 2.Neuner_2017\n",
      "2670   2 2.Neuner_2017\n",
      "2670   7 7.Zolochevska_2018\n",
      "7280   7 7.Zolochevska_2018\n",
      "23154   2 2.Neuner_2017\n",
      "23154   3 3.Sweet_2016\n",
      "112755   1 1.Völgyi_2017\n",
      "112755   3 3.Sweet_2016\n",
      "115827   2 2.Neuner_2017\n",
      "1653   0 1.Völgyi_2017\n",
      "3190   0 1.Völgyi_2017\n",
      "7284   2 2.Neuner_2017\n",
      "119   3 3.Sweet_2016\n",
      "4724   3 3.Sweet_2016\n",
      "26227   5 6.Chang_2013\n",
      "4723   7 7.Zolochevska_2018\n",
      "3708   2 2.Neuner_2017\n",
      "3708   8 8.Krivinko_2018\n",
      "30845   4 5.Chang_2015\n",
      "30845   5 6.Chang_2013\n",
      "6262   8 8.Krivinko_2018\n",
      "8831   2 2.Neuner_2017\n",
      "8831   8 8.Krivinko_2018\n",
      "4735   3 3.Sweet_2016\n",
      "4735   4 5.Chang_2015\n",
      "4735   8 8.Krivinko_2018\n",
      "1152   5 6.Chang_2013\n",
      "1152   7 7.Zolochevska_2018\n",
      "5245   8 8.Krivinko_2018\n",
      "57468   8 8.Krivinko_2018\n",
      "4741   4 5.Chang_2015\n",
      "4741   7 7.Zolochevska_2018\n",
      "134   2 2.Neuner_2017\n",
      "54918   2 2.Neuner_2017\n",
      "3208   3 3.Sweet_2016\n",
      "2697   1 1.Völgyi_2017\n",
      "4744   4 5.Chang_2015\n",
      "23176   4 5.Chang_2015\n",
      "23176   5 6.Chang_2013\n",
      "10376   7 7.Zolochevska_2018\n",
      "84617   7 7.Zolochevska_2018\n",
      "10382   4 5.Chang_2015\n",
      "10382   7 7.Zolochevska_2018\n",
      "1173   3 3.Sweet_2016\n",
      "6543   8 8.Krivinko_2018\n",
      "9369   2 2.Neuner_2017\n",
      "6812   7 7.Zolochevska_2018\n",
      "55964   8 8.Krivinko_2018\n",
      "161   2 2.Neuner_2017\n",
      "163   2 2.Neuner_2017\n",
      "7846   5 6.Chang_2013\n",
      "7846   7 7.Zolochevska_2018\n",
      "25769   2 2.Neuner_2017\n",
      "5802   2 2.Neuner_2017\n",
      "7345   5 6.Chang_2013\n",
      "7345   7 7.Zolochevska_2018\n",
      "3251   0 1.Völgyi_2017\n",
      "10935   6 6.Chang_2013\n",
      "2746   2 2.Neuner_2017\n",
      "2746   5 6.Chang_2013\n",
      "6843   1 1.Völgyi_2017\n",
      "6844   3 3.Sweet_2016\n",
      "1211   7 7.Zolochevska_2018\n",
      "2752   5 6.Chang_2013\n",
      "9922   2 2.Neuner_2017\n",
      "5315   5 6.Chang_2013\n",
      "6853   2 2.Neuner_2017\n",
      "6853   7 7.Zolochevska_2018\n",
      "6853   8 8.Krivinko_2018\n",
      "6854   2 2.Neuner_2017\n",
      "6854   4 5.Chang_2015\n",
      "1737   2 2.Neuner_2017\n",
      "3785   2 2.Neuner_2017\n",
      "203   3 3.Sweet_2016\n",
      "6857   6 6.Chang_2013\n",
      "1742   8 8.Krivinko_2018\n",
      "1743   2 2.Neuner_2017\n",
      "1743   4 5.Chang_2015\n",
      "4817   0 1.Völgyi_2017\n",
      "8402   8 8.Krivinko_2018\n",
      "2770   8 8.Krivinko_2018\n",
      "10452   1 1.Völgyi_2017\n",
      "2773   2 2.Neuner_2017\n",
      "54996   2 2.Neuner_2017\n",
      "2775   5 6.Chang_2013\n",
      "2776   3 3.Sweet_2016\n",
      "10971   3 3.Sweet_2016\n",
      "1759   7 7.Zolochevska_2018\n",
      "25824   7 7.Zolochevska_2018\n",
      "230   5 6.Chang_2013\n",
      "3306   3 3.Sweet_2016\n",
      "10989   2 2.Neuner_2017\n",
      "92399   5 6.Chang_2013\n",
      "3312   2 2.Neuner_2017\n",
      "3312   5 6.Chang_2013\n",
      "9456   3 3.Sweet_2016\n",
      "9456   8 8.Krivinko_2018\n",
      "100526833   8 8.Krivinko_2018\n",
      "7923   2 2.Neuner_2017\n",
      "4340   1 1.Völgyi_2017\n",
      "1267   3 3.Sweet_2016\n",
      "1267   7 7.Zolochevska_2018\n",
      "2806   2 2.Neuner_2017\n",
      "10487   2 2.Neuner_2017\n",
      "6900   4 5.Chang_2015\n",
      "2805   5 6.Chang_2013\n",
      "3315   6 6.Chang_2013\n",
      "7414   7 7.Zolochevska_2018\n",
      "7417   8 8.Krivinko_2018\n",
      "10493   4 5.Chang_2015\n",
      "50944   1 1.Völgyi_2017\n",
      "2819   0 1.Völgyi_2017\n",
      "2823   2 2.Neuner_2017\n",
      "2824   2 2.Neuner_2017\n",
      "3848   2 2.Neuner_2017\n",
      "3848   7 7.Zolochevska_2018\n",
      "3849   2 2.Neuner_2017\n",
      "5898   2 2.Neuner_2017\n",
      "5898   3 3.Sweet_2016\n",
      "7431   3 3.Sweet_2016\n",
      "2319   3 3.Sweet_2016\n",
      "1808   3 3.Sweet_2016\n",
      "1808   5 6.Chang_2013\n",
      "3857   7 7.Zolochevska_2018\n",
      "5906   1 1.Völgyi_2017\n",
      "3858   2 2.Neuner_2017\n",
      "3858   7 7.Zolochevska_2018\n",
      "4884   2 2.Neuner_2017\n",
      "794   0 1.Völgyi_2017\n",
      "28958   2 2.Neuner_2017\n",
      "287   2 2.Neuner_2017\n",
      "287   8 8.Krivinko_2018\n",
      "10015   5 6.Chang_2013\n",
      "4897   2 2.Neuner_2017\n",
      "286   8 8.Krivinko_2018\n",
      "292   8 8.Krivinko_2018\n",
      "5413   4 5.Chang_2015\n",
      "6950   2 2.Neuner_2017\n",
      "8997   8 8.Krivinko_2018\n",
      "811   7 7.Zolochevska_2018\n",
      "2861   2 2.Neuner_2017\n",
      "25902   2 2.Neuner_2017\n",
      "302   7 7.Zolochevska_2018\n",
      "815   7 7.Zolochevska_2018\n",
      "815   8 8.Krivinko_2018\n",
      "1329   2 2.Neuner_2017\n",
      "9522   1 1.Völgyi_2017\n",
      "9522   2 2.Neuner_2017\n",
      "816   8 8.Krivinko_2018\n",
      "348980   2 2.Neuner_2017\n",
      "308   6 6.Chang_2013\n",
      "57142   2 2.Neuner_2017\n",
      "80700   5 6.Chang_2013\n",
      "11069   3 3.Sweet_2016\n",
      "55102   2 2.Neuner_2017\n",
      "22848   1 1.Völgyi_2017\n",
      "1345   2 2.Neuner_2017\n",
      "11076   4 5.Chang_2015\n",
      "11076   5 6.Chang_2013\n",
      "2890   2 2.Neuner_2017\n",
      "2890   8 8.Krivinko_2018\n",
      "10059   2 2.Neuner_2017\n",
      "5962   3 3.Sweet_2016\n",
      "2892   3 3.Sweet_2016\n",
      "2892   8 8.Krivinko_2018\n",
      "2891   3 3.Sweet_2016\n",
      "2891   8 8.Krivinko_2018\n",
      "84303   2 2.Neuner_2017\n",
      "2893   3 3.Sweet_2016\n",
      "2893   8 8.Krivinko_2018\n",
      "9554   2 2.Neuner_2017\n",
      "9554   3 3.Sweet_2016\n",
      "80725   2 2.Neuner_2017\n",
      "80725   4 5.Chang_2015\n",
      "80725   8 8.Krivinko_2018\n",
      "2904   3 3.Sweet_2016\n",
      "7001   4 5.Chang_2015\n",
      "348   0 1.Völgyi_2017\n",
      "348   2 2.Neuner_2017\n",
      "351   2 2.Neuner_2017\n",
      "9568   1 1.Völgyi_2017\n",
      "2913   2 2.Neuner_2017\n",
      "2913   3 3.Sweet_2016\n",
      "2912   3 3.Sweet_2016\n",
      "2915   8 8.Krivinko_2018\n",
      "22884   4 5.Chang_2015\n",
      "5478   3 3.Sweet_2016\n",
      "7018   0 1.Völgyi_2017\n",
      "6506   1 1.Völgyi_2017\n",
      "85358   3 3.Sweet_2016\n",
      "22895   4 5.Chang_2015\n",
      "6512   2 2.Neuner_2017\n",
      "4978   1 1.Völgyi_2017\n",
      "126328   2 2.Neuner_2017\n",
      "1400   5 6.Chang_2013\n",
      "1400   6 6.Chang_2013\n",
      "55163   0 1.Völgyi_2017\n",
      "8573   8 8.Krivinko_2018\n",
      "8574   0 1.Völgyi_2017\n",
      "2944   2 2.Neuner_2017\n",
      "6529   1 1.Völgyi_2017\n",
      "2946   6 6.Chang_2013\n",
      "387   3 3.Sweet_2016\n",
      "388   3 3.Sweet_2016\n",
      "79751   2 2.Neuner_2017\n",
      "79751   8 8.Krivinko_2018\n",
      "391   4 5.Chang_2015\n",
      "396   5 6.Chang_2013\n",
      "396   7 7.Zolochevska_2018\n",
      "254863   2 2.Neuner_2017\n",
      "9615   5 6.Chang_2013\n",
      "219537   2 2.Neuner_2017\n",
      "10130   0 1.Völgyi_2017\n",
      "140690   2 2.Neuner_2017\n",
      "1428   0 1.Völgyi_2017\n",
      "1938   2 2.Neuner_2017\n",
      "51090   2 2.Neuner_2017\n",
      "22931   2 2.Neuner_2017\n",
      "22933   6 6.Chang_2013\n",
      "5019   5 6.Chang_2013\n",
      "22941   1 1.Völgyi_2017\n",
      "9118   4 5.Chang_2015\n",
      "6563   2 2.Neuner_2017\n",
      "2475   2 2.Neuner_2017\n",
      "10675   1 1.Völgyi_2017\n",
      "1462   4 5.Chang_2015\n",
      "5049   0 1.Völgyi_2017\n",
      "9145   2 2.Neuner_2017\n",
      "55737   2 2.Neuner_2017\n",
      "4540   2 2.Neuner_2017\n",
      "5052   3 3.Sweet_2016\n",
      "5052   5 6.Chang_2013\n",
      "51645   4 5.Chang_2015\n",
      "961   1 1.Völgyi_2017\n",
      "966   1 1.Völgyi_2017\n",
      "55752   4 5.Chang_2015\n",
      "55752   5 6.Chang_2013\n",
      "55753   2 2.Neuner_2017\n",
      "5582   8 8.Krivinko_2018\n",
      "3043   7 7.Zolochevska_2018\n",
      "4054   2 2.Neuner_2017\n",
      "6616   1 1.Völgyi_2017\n",
      "6616   2 2.Neuner_2017\n",
      "5594   2 2.Neuner_2017\n",
      "2011   7 7.Zolochevska_2018\n",
      "476   1 1.Völgyi_2017\n",
      "477   1 1.Völgyi_2017\n",
      "478   1 1.Völgyi_2017\n",
      "478   3 3.Sweet_2016\n",
      "479   2 2.Neuner_2017\n",
      "989   4 5.Chang_2015\n",
      "989   7 7.Zolochevska_2018\n",
      "481   2 2.Neuner_2017\n",
      "1499   8 8.Krivinko_2018\n",
      "10211   3 3.Sweet_2016\n",
      "5604   0 1.Völgyi_2017\n",
      "5093   4 5.Chang_2015\n",
      "998   3 3.Sweet_2016\n",
      "5095   2 2.Neuner_2017\n",
      "10728   3 3.Sweet_2016\n",
      "2023   5 6.Chang_2013\n",
      "2023   6 6.Chang_2013\n",
      "8682   0 1.Völgyi_2017\n",
      "490   1 1.Völgyi_2017\n",
      "23530   4 5.Chang_2015\n",
      "493   2 2.Neuner_2017\n",
      "2026   5 6.Chang_2013\n",
      "5621   2 2.Neuner_2017\n",
      "2554   2 2.Neuner_2017\n",
      "9212   2 2.Neuner_2017\n",
      "False\n",
      "True\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\homoe\\Miniconda3\\envs\\finalproject\\lib\\site-packages\\ipykernel_launcher.py:85: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entrez ID</th>\n",
       "      <th>1.Völgyi_2017</th>\n",
       "      <th>2.Neuner_2017</th>\n",
       "      <th>3.Sweet_2016</th>\n",
       "      <th>5.Chang_2015</th>\n",
       "      <th>6.Chang_2013</th>\n",
       "      <th>7.Zolochevska_2018</th>\n",
       "      <th>8.Krivinko_2018</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>93185</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4099</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>523</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>9228</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>526</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Entrez ID  1.Völgyi_2017  2.Neuner_2017  3.Sweet_2016  5.Chang_2015  \\\n",
       "0      93185              0              1             0             0   \n",
       "1       4099              0              1             0             0   \n",
       "2        523              0              0             0             0   \n",
       "3       9228              0              0             1             0   \n",
       "4        526              0              1             0             0   \n",
       "\n",
       "   6.Chang_2013  7.Zolochevska_2018  8.Krivinko_2018  count  \n",
       "0             0                   0                0      1  \n",
       "1             0                   0                0      1  \n",
       "2             1                   0                0      1  \n",
       "3             0                   0                0      1  \n",
       "4             0                   0                0      1  "
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# creating a single list with all of them\n",
    "\n",
    "DE_p_2 = [[],[], [], [], [], [], [], [], [] ]\n",
    "import math\n",
    "# DE_proteins_unique = []\n",
    "# for i in range(len(DE_proteins)):\n",
    "#     for j in range(len(DE_proteins[i][0])):\n",
    "#         if(not (math.isnan(DE_proteins[i][0][j]))):\n",
    "#             DE_p_2[i].append(str(int(DE_proteins[i][0][j])))\n",
    "#             if( DE_proteins[i][0][j] not in DE_proteins_unique):\n",
    "#                 DE_proteins_unique.append(str(int(DE_proteins[i][0][j])))\n",
    "#                 DE_proteins_unique.append(DE_proteins[i][0][j])\n",
    "                \n",
    "#             DE_proteins[i][0][j] = str(int(DE_proteins[i][0][j]))\n",
    "\n",
    "                #           DE_  print(str(int(DE_proteins[i][0][j])))\n",
    "    \n",
    "DE_proteins_un = np.array([])\n",
    "for i in range(len(DE_proteins)):\n",
    "#     print(DE_proteins[i][0])\n",
    "    DE_proteins_un = np.concatenate((DE_proteins_un,DE_proteins[i][0]))\n",
    "\n",
    "DE_proteins_unique = np.array(list(set(DE_proteins_un)),dtype=int)\n",
    "\n",
    "# print(len(DE_proteins_un))\n",
    "# print(len(DE_proteins_unique))\n",
    "print(len(set(DE_proteins_unique)))\n",
    "# print(DE_proteins_unique)\n",
    "\n",
    "\n",
    "# checking in each making df\n",
    "\n",
    "\n",
    "colnames = ['Entrez ID']\n",
    "\n",
    "for i in range(len(filelist)):\n",
    "    name = filelist[i].split('\\n')[0]\n",
    "    colname = name.split('/')[0]\n",
    "    if(colname not in colnames):\n",
    "        colnames.append(colname)\n",
    "\n",
    "# print(colnames)\n",
    "print(len(filelist))\n",
    "print(len(DE_proteins))\n",
    "\n",
    "DE_df = pd.DataFrame(columns=colnames)\n",
    "\n",
    "for i in range(len(DE_proteins_unique)):\n",
    "    pro = DE_proteins_unique[i]\n",
    "    newrow = {\n",
    "        'Entrez ID': pro,\n",
    "        '1.Völgyi_2017': 0,\n",
    "        '2.Neuner_2017': 0,\n",
    "        '3.Sweet_2016': 0,\n",
    "        '5.Chang_2015': 0,\n",
    "        '6.Chang_2013': 0,\n",
    "        '7.Zolochevska_2018': 0,\n",
    "        '8.Krivinko_2018': 0\n",
    "    }\n",
    "    DE_df = DE_df.append(newrow,ignore_index=True)\n",
    "    \n",
    "    names = ['1.Völgyi_2017','1.Völgyi_2017','2.Neuner_2017','3.Sweet_2016','5.Chang_2015','6.Chang_2013','6.Chang_2013','7.Zolochevska_2018','8.Krivinko_2018']\n",
    "    for j in range(len(DE_proteins)):\n",
    "        col=names[j]\n",
    "        if(pro in DE_proteins[j][0]):\n",
    "            print(pro, ' ', j,col )\n",
    "            DE_df.loc[DE_df['Entrez ID'] == pro, col] = 1\n",
    "        \n",
    "#     for j in range(len(filelist)):\n",
    "#         name = filelist[j].split('\\n')[0]\n",
    "#         colname = name.split('/')[0]\n",
    "# #         print(colname)\n",
    "# #         print(DE_proteins[j][0])\n",
    "# #         print(pro, int(pro) in DE_proteins[j][0], j, type(pro))\n",
    "#         if(pro in DE_p_2[j]):\n",
    "# #             if(j == 0):\n",
    "# #                 print('in')\n",
    "# #             print( pro, ' ', colname)\n",
    "#             DE_df.loc[DE_df['Entrez ID'] == pro, colname] = 1\n",
    "\n",
    "\n",
    "DE_df['count'] = DE_df['1.Völgyi_2017'] + DE_df['2.Neuner_2017'] + DE_df['3.Sweet_2016'] + DE_df['5.Chang_2015'] + DE_df['6.Chang_2013'] + DE_df['7.Zolochevska_2018'] + DE_df['8.Krivinko_2018']\n",
    "\n",
    "\n",
    "print( '2167' in DE_proteins_unique)\n",
    "print( 2617 in DE_proteins_unique)\n",
    "DE_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "a\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "276"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if( 2617 in DE_proteins[0][0]):\n",
    "    print('a')\n",
    "    \n",
    "DE_df.sort_values('count')\n",
    "# DE_df.tail(100)\n",
    "len(DE_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2617.0\n",
      "True\n",
      "True\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "276"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p = float(2617)\n",
    "print(p)\n",
    "print( p in DE_proteins[0][0])\n",
    "print( 2617.0 == DE_proteins[0][0][0])\n",
    "# DE_df.loc[DE_df['Entrez ID'] == '2617','2.Neuner_2017' ] = 1\n",
    "# DE_proteins_unique\n",
    "DE_df.head()\n",
    "# DE_proteins[0][0]\n",
    "# DE_df.sort_values('count', ascending=False)\n",
    "DE_df.to_csv('Diffrentially_Expressed.csv',index=False)\n",
    "\n",
    "DE_df.sort_values(by='count')\n",
    "len(list(set(DE_proteins_unique)))\n",
    "# len(DE_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "done\n"
     ]
    }
   ],
   "source": [
    "DEproteinslist = []\n",
    "DEproteinslist = pd.DataFrame()\n",
    "DEproteinslist['Entrez id'] = DE_df['Entrez ID']\n",
    "DEproteinslist['A'] = 'Diff_ex'\n",
    "DEproteinslist['B'] = 'Differentiallly expressed proteins'\n",
    "# print(len(DEproteinslist))\n",
    "# print(len(DE_df['Entrez ID']))\n",
    "# print(len(DE_df))\n",
    "# DEproteinslist\n",
    "# DE_df.head(450)\n",
    "# print(DEproteinslist)\n",
    "# DEproteinslist.to_csv('flatfile.DE.csv',sep='\\t',header=False, columns=['A','B','Entrez id'],index=False)\n",
    "print('done')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Entrez ID' '1.Völgyi_2017' '2.Neuner_2017' '3.Sweet_2016' '5.Chang_2015'\n",
      " '6.Chang_2013' '7.Zolochevska_2018' '8.Krivinko_2018' 'count']\n"
     ]
    }
   ],
   "source": [
    "DEproteins3list = []\n",
    "DEproteins3 = pd.DataFrame()\n",
    "DEproteins_3 = pd.DataFrame()\n",
    "\n",
    "DEproteins2list = []\n",
    "DEproteins2 = pd.DataFrame()\n",
    "DEproteins_2 = pd.DataFrame()\n",
    "\n",
    "DE_df.sort_values(by='count')\n",
    "print(DE_df.columns.values)\n",
    "DEproteins  = DE_df[['Entrez ID','count']]\n",
    "DEproteins3 = DEproteins[DEproteins['count'] == 3]\n",
    "DEproteins_3['Entrez id'] = DEproteins3['Entrez ID']\n",
    "DEproteins_3['A'] = 'Diff_ex_3'\n",
    "DEproteins_3['B'] = 'Differentially expressed proteins in 3'\n",
    "# DEproteins3['Entrez id'] == \n",
    "\n",
    "DEproteins2 = DEproteins[DEproteins['count'] == 2]\n",
    "DEproteins_2['Entrez id'] = DEproteins2['Entrez ID']\n",
    "DEproteins_2['A'] = 'Diff_ex_2'\n",
    "DEproteins_2['B'] = 'Differentially expressed proteins in 2'\n",
    "\n",
    "DEproteins_2.head()\n",
    "\n",
    "DEproteinslist = DEproteinslist.append(DEproteins_3,ignore_index=True )\n",
    "DEproteinslist = DEproteinslist.append(DEproteins_2,ignore_index=True )\n",
    "DEproteinslist['A'].unique()\n",
    "# DEproteinslist\n",
    "\n",
    "DEproteinslist.to_csv('flatfile.DE.sep.csv',sep='\\t',header=False, columns=['A','B','Entrez id'],index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "DEproteins_2.head()\n",
    "\n",
    "DEproteins_2.to_csv('Diff_exp2.csv',sep='\\t', header=False, columns=['Entrez id'],index=False)\n",
    "DEproteins_3.to_csv('Diff_exp3.csv',sep='\\t', header=False, columns=['Entrez id'],index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "finalproject",
   "language": "python",
   "name": "finalproject"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
